lib/ret_web/controllers/api/v1/avatar_controller.ex

```diff
@@ -2,45 +2,32 @@ defmodule RetWeb.Api.V1.AvatarController do
   use RetWeb, :controller
 
   alias Ret.{Account, Repo, Avatar, AvatarListing, Storage, GLTFUtils}
-  import Ecto.Query
 
   plug(RetWeb.Plugs.RateLimit when action in [:create, :update])
 
-  defp get_avatar(avatar_sid) do
+  defp get_avatar(avatar_sid, preloads \\ []) do
     avatar_sid
     |> Avatar.avatar_or_avatar_listing_by_sid()
-    |> preload()
+    |> preload(preloads)
   end
 
-  defp preload(nil = _avatar) do
-    nil
+  defp preload(a) do
+    preload(a, [])
   end
 
-  defp preload(%Avatar{} = a) do
-    a |> Repo.preload([Avatar.file_columns() ++ [:parent_avatar, :parent_avatar_listing, :account]])
+  defp preload(%Avatar{} = a, preloads) do
+    a |> Repo.preload([Avatar.file_columns() ++ [:parent_avatar, :parent_avatar_listing, :avatar_listings]])
```

Comment:
did you intentionally drop `account` here?

Reply:
Yep. We only ever look at the avatar's account_id so no need to load the whole association.